src/compliance.rs

```diff
@@ -0,0 +1,293 @@
+                                checks.pr_dismiss_stale = true;
+                            }
+                            if params
+                                .get("require_code_owner_review")
+                                .and_then(|v| v.as_bool())
+                                .unwrap_or(false)
+                            {
+                                checks.pr_require_code_owner = true;
+                            }
+                        }
+                    }
+                    "required_status_checks" => {
+                        if let Some(params) = rule.parameters.as_ref() {
+                            if params
+                                .get("required_status_checks")
+                                .and_then(|v| v.as_array())
+                                .map(|a| !a.is_empty())
+                                .unwrap_or(false)
+                            {
+                                checks.require_status_checks = true;
+                            }
+                        }
+                    }
+                    _ => {}
+                }
+            }
+        }
+
+        print_report(&repo, &default_branch, checks);
+    }
+}
+
+fn print_report(repo: &str, branch: &str, checks: ProtectionChecks) {
+    println!(
+        "{} {}  {} {}  {} {}",
+        "Repo:".yellow(),
+        repo.white(),
+        "Default branch:".yellow(),
+        branch.white(),
+        "Score (0-7):".yellow(),
+        checks.score().to_string().white()
+    );
+    println!(
+        "  - PR requires one approval: {}\n  - PR: dismiss stale reviews: {}\n  - PR requires code owners approval: {}\n  - Force-push disabled: {}\n  - Deletion disabled: {}\n  - Require signed commits: {}\n  - Require status checks: {}\n",
+        check_symbol(checks.pr_one_approval),
+        check_symbol(checks.pr_dismiss_stale),
+        check_symbol(checks.pr_require_code_owner),
+        check_symbol(checks.disable_force_push),
+        check_symbol(checks.disable_deletion),
+        check_symbol(checks.require_signed_commits),
+        check_symbol(checks.require_status_checks)
+    );
+}
+
+fn check_symbol(v: bool) -> String {
+    if v {
+        return "✅".green().to_string();
+    }
+    "❌".red().to_string()
+}
+
+fn get_default_branch(bootstrap: &Bootstrap, repo: &str) -> Option<String> {
+    // Try repository metadata first
+    if let Ok(res) = make_github_request(
+        &bootstrap.token,
+        &format!("/repos/{}/{repo}", bootstrap.org),
+        3,
+        None,
+    ) {
+        if let Some(branch) = res.get("default_branch").and_then(|v| v.as_str()) {
+            return Some(branch.to_string());
+        }
+    }
+
+    // Fallbacks: probe common default branch names
+    if branch_exists(bootstrap, repo, "main") {
+        return Some("main".to_string());
+    }
+    if branch_exists(bootstrap, repo, "master") {
+        return Some("master".to_string());
+    }
+    None
```

Comment:
We shouldn't fallback because we could be wrong. We should just fail

Reply:
agree, removed.